kivy/core/window/__init__.py

```diff
@@ -629,11 +629,31 @@ def _get_android_kheight(self):
             import android
         return android.get_keyboard_height()
 
+    def _get_kivy_vkheight(self):
+        mode = Config.get('kivy', 'keyboard_mode')
+        smode = self.softinput_mode
+        if (
+            (mode == 'dock' or mode == 'systemanddock')
+            and self._vkeyboard_cls is not None
+        ):
+            for w in self.children:
+                if isinstance(w, VKeyboard):
+                    vkeyboard_height = w.height * w.scale
+                    if smode == 'pan':
+                        w.top = 0
```

Comment:
Can we avoid assigning `top` property in this method as it should only return the height of virtual keyboard?

Reply:
@pythonic64 Okay, so I think the code snippet below would be a better alternative:

```python
def _set_vkeyboard_top(self, vkeyboard, top=0):
    vkeyboard.top = top

...
if isinstance(w, VKeyboard):
    vkeyboard_height = w.height * w.scale
    if smode == 'below_target':
        self._set_vkeyboard_top(w, w.target.y)
        return vkeyboard_height - w.target.y
...


    

```

Reply:
What I meant is that `_get_kivy_vkheight` (and any other getter) should not assign any values, but only get the requested value, in this case a height of virtual keyboard. 

Object opening the keyboard widget should assign its position.

Reply:
You are right! My last commit fixes this.

Reply:
Looking good 👍